gobblin-api/src/main/java/gobblin/source/workunit/ImmutableWorkUnit.java

```diff
@@ -20,11 +27,49 @@
 public class ImmutableWorkUnit extends WorkUnit {
 
   public ImmutableWorkUnit(WorkUnit workUnit) {
-    super(workUnit);
+    super(workUnit.getExtract());
```

Comment:
Seems the `Extract` object in an `ImmutableWorkUnit` is still mutable. Is this acceptable?